src/lib389/lib389/plugins.py

```diff
@@ -827,6 +827,56 @@ def _lint_member_attr_indexes(self):
                         report['check'] = f'memberof:attr_indexes'
                         yield report
 
+    def _lint_member_substring_index(self):
+        if self.status():
+            from lib389.backend import Backends
+            backends = Backends(self._instance).list()
+            attrs = self.get_attr_vals_utf8_l("memberofgroupattr")
+            container = self.get_attr_val_utf8_l("nsslapd-plugincontainerscope")
+            #import pdb; pdb.set_trace()
+            for backend in backends:
+                suffix = backend.get_attr_val_utf8_l('nsslapd-suffix')
+                if suffix == "cn=changelog":
+                    # Always skip retro changelog
+                    continue
+                if container is not None:
+                    # Check if this backend is in the scope
+                    if not container.endswith(suffix):
+                        # skip this backend that is not in the scope
+                        continue
+                indexes = backend.get_indexes()
+                membership_attrs = ['member', 'uniquemember']
+                for attr in membership_attrs:
+                    report = copy.deepcopy(DSMOLE0002)
+                    try:
+                        index = indexes.get(attr)
+                        types = index.get_attr_vals_utf8_l("nsIndexType")
+                        valid = True
+                        if "sub" in types:
+                            valid = False
+
+                        if not valid:
+                            report['detail'] = report['detail'].replace('ATTR', attr)
+                            report['detail'] = report['detail'].replace('BACKEND', suffix)
+                            report['fix'] = report['fix'].replace('ATTR', attr)
+                            report['fix'] = report['fix'].replace('BACKEND', suffix)
+                            report['fix'] = report['fix'].replace('YOUR_INSTANCE', self._instance.serverid)
+                            report['items'].append(suffix)
+                            report['items'].append(attr)
+                            report['check'] = f'attr:substring_index'
+                            yield report
+                    except:
+                        # No index at all, bad
```

Comment:
If it's a 'No index at all' error, I think, we should be more specific with the `except:` and specify the exact exception expected.

Reply:
I  copied this from other memberof attr checks, but you're right. This test is to check if a membership attribute contains a substring index, if the index doesn't exist then we have nothing to report.